specification/voiceservices/resource-manager/Microsoft.VoiceServices/cadl/communicationsgateway.cadl

```diff
@@ -0,0 +1,196 @@
+// Definitions for the CommunicationsGateway resource and child resources
+
+import "@cadl-lang/openapi";
+import "@cadl-lang/rest";
+import "@azure-tools/cadl-providerhub";
+import "@azure-tools/cadl-autorest";
+import "@azure-tools/cadl-azure-core";
+import "@azure-tools/cadl-azure-resource-manager";
+
+namespace Microsoft.VoiceServices;
+
+using Cadl.Http;
+using Cadl.Rest;
+using Azure.ResourceManager;
+
+@doc("A TestLine resource")
+@parentResource(CommunicationsGateway)
+model TestLine is TrackedResource<TestLineProperties> {
+  @pattern("^[a-zA-Z0-9-]{3,24}$")
+  @key("testLineName")
+  @segment("testLines")
+  @doc("Unique identifier for this test line")
+  @visibility("Read")
+  @path
+  name: string;
+}
+
+@doc("The purpose of the TestLine resource.")
+enum TestLinePurpose {
+  Manual,
+  Automated,
+}
+
+@doc("Details of the TestLine resource.")
+model TestLineProperties is ResourceProperties {
+  @doc("The phone number")
+  @visibility("read", "write", "create")
+  phoneNumber: string;
+  @doc("Purpose of this test line, e.g. automated or manual testing")
+  @visibility("read", "write", "create")
+  purpose: TestLinePurpose;
+}
+
+@armResourceOperations
+interface TestLines extends ResourceRead<TestLine>, ResourceCreate<TestLine>, ResourceDelete<TestLine>, ResourceListByParent<TestLine>, ResourceUpdateNoProperties<TestLine> {}
+
+@doc("A Contact resource")
+@parentResource(CommunicationsGateway)
+model Contact is TrackedResource<ContactProperties> {
+  @pattern("^[a-zA-Z0-9-]{3,24}$")
+  @key("contactName")
+  @segment("contacts")
+  @doc("Unique identifier for this contact")
+  @visibility("Read")
+  @path
+  name: string;
+}
+
+@doc("Details of the Contact resource.")
+model ContactProperties is ResourceProperties {
+  @doc("Full name of contact")
+  @visibility("read", "write", "create")
+  contactName: string;
+  @doc("Telephone number of contact")
+  @visibility("read", "write", "create")
+  phoneNumber: string;
+  @doc("Email address of contact")
+  @visibility("read", "write", "create")
+  email: string;
+  @doc("Job title of contact")
+  @visibility("read", "write", "create")
+  role: string;
+}
+
+@armResourceOperations
+interface Contacts extends ResourceRead<Contact>, ResourceCreate<Contact>, ResourceDelete<Contact>, ResourceListByParent<Contact>, ResourceUpdateNoProperties<Contact> {}
+
+@doc("A CommunicationsGateway resource")
+model CommunicationsGateway is TrackedResource<CommunicationsGatewayProperties> {
+  @pattern("^[a-zA-Z0-9-]{3,24}$")
+  @key("communicationsGatewayName")
+  @segment("communicationsGateways")
+  @doc("Unique identifier for this deployment")
+  @visibility("Read")
+  @path
+  name: string;
+}
+
+@doc("The status of the current CommunicationsGateway resource.")
+enum Status {
```

Comment:
>Status [](http://example.com/codeflow?start=5&length=6)

This is not modeled as string? Consider if this can be modeled as string for future extensibility

Reply:
This PR is just copying https://github.com/Azure/azure-rest-api-specs-pr/blob/RPSaaSMaster/specification/voiceservices/resource-manager/Microsoft.VoiceServices/preview/2022-12-01-preview/openapi.json across to the public repo (which has been approved - see the PR link above), so I don't think we want to change the API at this point? But this is something we're looking at changing before we stabilize the API for GA.

Reply:
Sounds good